File: 3838-weighted-word-mapping/3838-weighted-word-mapping.c
char* mapWordWeights(char** words, int n, int* weights, int weightsSize) {
    int i = 0;
    char* ans = (char*)malloc(sizeof(char) * (n + 1));
    char* help = "abcdefghijklmnopqrstuvwxyz";
    for (; i < n; i++) {
        int s = 0;
        char* str = words[i];
        for (; *str; str++)
            s += weights[(*str - 'a')];

        int r = 25 - (s % 26);
        ans[i] = help[r];
    }
    ans[i] = '\0';
    return ans;
}

File: 3838-weighted-word-mapping/3838-weighted-word-mapping.go
func mapWordWeights(words []string, weights []int) string {
	n := len(words)
	ans := make([]byte, n)

	for i, word := range words {
		sum := 0
		for _, ch := range word {
			sum += weights[ch-'a']
		}

		r := 25 - (sum % 26)
		ans[i] = byte('a' + r)
	}

	return string(ans)
}